### PR TITLE
feat: created at updated at from block time

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
Previously before this PR, users submitting documents were expected to set the document created at and updated at time on the client side. Then Platform would validate in consensus that these values were close to to the current block time. This approach had a few downsides. First it meant that documents in mempool would expire even though users did nothing wrong. The second is that it required users to have a clock that was very close in time to the network time.

## What was done?

Now created at and updated at are set by the block time. Meaning that they are automatic and users have nothing to set manually.

## How Has This Been Tested?
Test suite is passing

## Breaking Changes
This is a breaking change.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added or updated relevant unit/integration/functional/e2e tests
- [X] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [X] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_create_transition/v0/mod.rs [hunk 1]
```diff
@@ -268,8 +226,8 @@ impl DocumentFromCreateTransitionV0 for Document {
                         owner_id,
                         properties: data,
                         revision: document_type.initial_revision(),
-                        created_at,
-                        updated_at,
+                        created_at: block_time,
```
Comment (shumkov) [inline packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_create_transition/v0/mod.rs:229]: Only add these values if required

File: packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_create_transition/v0/mod.rs [hunk 2]
```diff
@@ -320,8 +273,8 @@ impl DocumentFromCreateTransitionV0 for Document {
                         owner_id,
                         properties: data.clone(),
                         revision: document_type.initial_revision(),
-                        created_at: *created_at,
-                        updated_at: *updated_at,
+                        created_at: block_time,
```
Comment (shumkov) [inline packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_create_transition/v0/mod.rs:276]: Only add these values if required

File: packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_replace_transition/v0/mod.rs [hunk 3]
```diff
@@ -327,7 +316,7 @@ impl DocumentFromReplaceTransitionV0 for Document {
                 properties: data.clone(),
                 revision: Some(*revision),
                 created_at,
-                updated_at: *updated_at,
+                updated_at: block_time,
```
Comment (shumkov) [inline packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_replace_transition/v0/mod.rs:319]: Set only if required

File: packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_replace_transition/v0/mod.rs [hunk 4]
```diff
@@ -364,7 +353,7 @@ impl DocumentFromReplaceTransitionV0 for Document {
                 properties: data,
                 revision: Some(revision),
                 created_at,
-                updated_at,
+                updated_at: block_time,
```
Comment (shumkov) [inline packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_replace_transition/v0/mod.rs:356]: Set only if required

**Review by shumkov — CHANGES_REQUESTED** [review]

**Comment by shumkov** [comment]
@QuantumExplorer feel free to merge it when you resolve my comments.

**Review by shumkov — APPROVED** [review]
@QuantumExplorer My approval in advance if you gonna be a good boy.

File: packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_create_transition/v0/mod.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_create_transition/v0/mod.rs]: Fixed

**Pull request merged**